exist-core/src/main/java/org/exist/xquery/functions/util/ModuleInfo.java

```diff
@@ -223,14 +223,19 @@ public Sequence eval(Sequence[] args, Sequence contextSequence)
 		} else {
 			final ValueSequence resultSeq = new ValueSequence();
             final XQueryContext tempContext = new XQueryContext(context.getBroker().getBrokerPool());
-			for(final Iterator<Module> i = tempContext.getRootModules(); i.hasNext(); ) {
-				final Module module = i.next();
-				resultSeq.add(new StringValue(this, module.getNamespaceURI()));
-			}
-			if (tempContext.getRepository().isPresent()) {
-			    for (final URI uri : tempContext.getRepository().get().getJavaModules()) {
-				resultSeq.add(new StringValue(this, uri.toString()));
-			    }
+			try {
+				for (final Iterator<Module> i = tempContext.getRootModules(); i.hasNext(); ) {
+					final Module module = i.next();
+					resultSeq.add(new StringValue(this, module.getNamespaceURI()));
+				}
+				if (tempContext.getRepository().isPresent()) {
+					for (final URI uri : tempContext.getRepository().get().getJavaModules()) {
+						resultSeq.add(new StringValue(this, uri.toString()));
+					}
+				}
+			} finally {
```

Comment:
in other code these two lines have different order. Is that on purpose? or does it not matter?

Reply:
Order here is not especially important